packages/visx-xychart/src/hooks/useScales.ts

```diff
@@ -42,8 +41,11 @@ export default function useScales<
 
     const xDomain = isDiscreteScale(xScaleConfig) ? xValues : d3Extent(xValues);
 
-    xScale.range(xScaleConfig.range || [xMin, xMax]);
-    xScale.domain(xScaleConfig.domain || xDomain);
+    let xScale = createScale({
+      range: [xMin, xMax],
+      domain: xDomain as [number, number],
```

Comment:
I think depending on scale type this won't always be a `number`, but I can try to fix the typing in a followup PR as types can be tricky to debug here.